SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialConnectionManager.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/Connection/SpatialOSWorkerInterface.h"
+#include "Interop/Connection/ConnectionConfig.h"
+#include "SpatialCommonTypes.h"
+#include "SpatialGDKSettings.h"
+
+#include "SpatialConnectionManager.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialConnectionManager, Log, All);
+
+class USpatialWorkerConnection;
+
+enum class ESpatialConnectionType
+{
+	Receptionist,
+	LegacyLocator,
+	Locator,
+	DevAuthFlow
+};
+
+UCLASS()
+class SPATIALGDK_API USpatialConnectionManager : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	virtual void FinishDestroy() override;
+	void DestroyConnection();
+	
+	using LoginTokenResponseCallback = TFunction<bool(const Worker_Alpha_LoginTokensResponse*)>;
+    
+    /// Register a callback using this function.
+    /// It will be triggered when receiving login tokens using the development authentication flow inside SpatialWorkerConnection.
+    /// @param Callback - callback function.
+	void RegisterOnLoginTokensCallback(const LoginTokenResponseCallback& Callback) {LoginTokenResCallback = Callback;}
+
+	void Connect(bool bConnectAsClient, uint32 PlayInEditorID);
+
+	FORCEINLINE bool IsConnected() { return bIsConnected; }
+
+	PhysicalWorkerName GetWorkerId() const;
+	const TArray<FString>& GetWorkerAttributes() const;
```

Comment:
Do we need these functions in `SpatialConnectionManager`?

Reply:
I removed those and pointed a couple of places at the connection's implementation.